libse/Utilities.cs

```diff
@@ -1702,19 +1702,31 @@ public static string RemoveSpaceBetweenNumbers(string text)
         /// <returns>text with unneeded spaces removed</returns>
         public static string RemoveUnneededSpaces(string text, string language)
         {
-            const string zeroWidthSpace = "\u200B";
-            const string zeroWidthNoBreakSpace = "\uFEFF";
-            const string noBreakSpace = "\u00A0";
-            const string operatingSystemCommand = "\u009D";
+            const char zeroWidthSpace = '\u200B';
+            const char zeroWidthNoBreakSpace = '\uFEFF';
+            const char noBreakSpace = '\u00A0';
+            const char operatingSystemCommand = '\u009D';
 
             text = text.Trim();
-
-            text = text.Replace(zeroWidthSpace, string.Empty);
-            text = text.Replace(zeroWidthNoBreakSpace, string.Empty);
-            text = text.Replace(noBreakSpace, " ");
-            text = text.Replace(operatingSystemCommand, string.Empty);
-            text = text.Replace('\t', ' ');
-
+            int len = text.Length;
+            int count = 0;
+            char[] textChars = new char[len];
+            for (int i = 0; i < len; i++)
+            {
+                char ch = text[i];
+                // Ignore: \u200B, \uFEFF and \u009D.
+                if (!(ch == zeroWidthSpace || ch == zeroWidthNoBreakSpace || ch == operatingSystemCommand))

```

Comment:
Well, about the switch I'm not sure it it's a good idea, note that here we are filtering (skipping) these chars not **selecting** then which would be better to do which a switch-statement.